files/models.py

```
@@ -1318,25 +1123,69 @@ def url(self):
+                    # 5) Rename physical path & FileField from .srt -> .vtt
+                    if file_lower.endswith(".srt"):
+                        new_path = input_path[:-4] + ".vtt"
+                        if new_path != input_path:
+                            os.rename(input_path, new_path)
+                        self.subtitle_file.name = self.subtitle_file.name[:-4] + ".vtt"
+                        self.save(update_fields=["subtitle_file"])
+                else:
+                    raise Exception("FFmpeg conversion failed - no output file created")
+            except Exception as e:
+                log.error(f"Subtitle conversion failed for {input_path}: {str(e)}")
+                raise Exception(f"Could not convert SRT file to VTT format: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)